### PR TITLE
PR 430 + Generator Instances + Monster Health fix

### PR DESCRIPTION
**Pull request by LtRipley36706**
PR ready for merge consideration...

This PR requires ACE-World v0.1.4 to work as expected.

* Rolled in PR #430 from @Lidefeath as starting base. 
* Added generator links table to WorldBase.
* Made changes to GuidManger and ObjectGuid in an effort to avoid collisions with guids already present in ACE-World. Does not persist or track, resets each server start.
* Above changes made IsCreature function, and smite works for at least the Sparring Golems found in the Academy, does not seem to work for Drudges in Holtburg.
* Speaking of Drudges, there is a drudge invasion south and west of Holtburg.
* Several changes to 06-06-30-2017-generator-chains-testdata.sql found in Database/Updates/World to set up for generator instances and show examples of them.
* Changed the way generators are classified to make use of some AceObject properties.
* Slightly adjusted debug output on ID panel.

### CODE REVIEW

**Comment by LtRipley36706** [comment]
[See this PR in ACE-World](https://github.com/ACEmulator/ACE-World/pull/24) for further examples of adding generators into the database

**Comment by LtRipley36706** [comment]
PR ready to be reviewed and considered for merging

File: Source/ACE/Factories/GeneratorFactory.cs [hunk 1]
```diff
@@ -85,38 +85,65 @@ public static List<WorldObject> CreateWorldObjectsFromGenerator(AceObject genera
...
+                    {
+                        results.Add(new Generator(new ObjectGuid(GuidManager.NewItemGuid()), generator));
+                        generator.GeneratorEnteredWorld = true;
+                    }
+
+                    // Get a random generator from the weighted list of linked generators and read it's AceObject from the DB
+                    if (generator.GeneratorLinks.Count == 0)
+                        return null;
+                    uint linkId = GetRandomGeneratorIdFromGeneratorList(random, generator.GeneratorLinks);
+                    var newGen = DatabaseManager.World.GetAceObjectByWeenie(linkId);
```
Comment (ddevec) [inline Source/ACE/Factories/GeneratorFactory.cs]: IMO var should only be used when the type is defined as part of the statements rvalue:

eg: var myThing = new ReallyLongClassNameThatsAnnoyingToType();

Not blocking PR over it, but please change.

File: Source/ACE/Factories/GeneratorFactory.cs [hunk 2]
```diff
@@ -85,38 +85,65 @@ public static List<WorldObject> CreateWorldObjectsFromGenerator(AceObject genera
...
+                    newGen.Generator = generator.AceObjectId;
+                    newGen.GeneratorEnteredWorld = true;
 
-                    default:
-                        baseObject.Location = pos;
-                        if (baseObject.Location != null)
-                            results.Add(new DebugObject(new ObjectGuid(baseObject.AceObjectId), baseObject));
-                        break;
+                    // Recursively call this method again with the just read generatorObject
+                    var objectList = CreateWorldObjectsFromGenerator(newGen);
```
Comment (ddevec) [inline Source/ACE/Factories/GeneratorFactory.cs]: improper use of var (see above)

**Pull request merged**